torchaudio/functional/functional.py

```diff
@@ -84,6 +84,14 @@ def spectrogram(
         ``n_fft // 2 + 1`` and ``n_fft`` is the number of
         Fourier bins, and time is the number of window hops (n_frame).
     """
+    if power is None and not return_complex:
+        warnings.warn(
```

Comment:
I think we should explicitly mention it's a deprecation warning: 

"Deprecation warning: In a future TorchAudio release `spectrogram` ",
"will no longer return tensors of pseudo complex type (..., 2) by default. ",
"Instead, `return_complex` will be set to `True` by default and spectrogram",
"would return native complex tensors by default."

Reply:
actually even better, we could use https://docs.python.org/3/library/exceptions.html#DeprecationWarning

`warnings.warn(msg, category=DeprecationWarning)` where `msg` equals
```
"In a future TorchAudio release, `spectrogram` ",
"will no longer return tensors of pseudo complex type (..., 2) by default. ",
"Instead, `return_complex` will be set to `True` by default and spectrogram",
"would return native complex tensors by default."
```

Reply:
@anjali411 
I do not have a preference here but, according to [this comment](https://github.com/pytorch/audio/pull/523#discussion_r405673957) by @gchanan, PyTorch uses `UserWarning` for deprecation, so in `torchaudio` we used `UserWarning`.

However, I see mixed usage in pytorch.
https://github.com/pytorch/pytorch/search?q=DeprecationWarning 
https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/torch/hub.py#L512

Reply:
@anjali411 Do we want to deprecate the pseudo complex in the upcoming release?

The planned migrations phases are as following and I was thinking to deprecate the native complex type in phase 2.
1. Add support for native complex type
2. Switch to native complex type by default
3. Remove pseudo complex type

Reply:
I gave some thoughts on this and now I think it makes more sense to give heads up on deprecation prior to defaulting native complex type.